src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -86,20 +86,51 @@ protected override string GetFilterText(ISymbol symbol, string displayText, CSha
             CompletionOptions options,
             CancellationToken cancellationToken)
         {
-            if (context != null && options.TriggerInArgumentLists)
+            if (context is null)
+                return true;
+
+            // Avoid preselection & hard selection when triggered via insertion in an argument list.
+            // If an item is hard selected, then a user trying to type MethodCall() will get
+            // MethodCall(someVariable) instead. We need only soft selected items to prevent this.
+            return !await IsTriggeredInArgumentListAsync(context, position, options, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected override async Task<bool> ShouldProvideAvailableSymbolsInCurrentContextAsync(
+            CompletionContext? completionContext,
+            CSharpSyntaxContext context,
+            int position,
+            CompletionOptions options,
+            CancellationToken cancellationToken)
+        {
+            if (completionContext is null)
+                return true;
+
+            // If we are triggered in argument list, provide symbols only when the invoked method accept any arguments.
+            // Special case for argument of base record primary constructor as a workaround for bug https://github.com/dotnet/roslyn/issues/70803
+            return !await IsTriggeredInArgumentListAsync(completionContext, position, options, cancellationToken).ConfigureAwait(false)
+                || (!context.InferredTypes.IsEmpty || IsInPrimaryConstructorArgumentList());
+
+            bool IsInPrimaryConstructorArgumentList()
+                => context.TargetToken.Parent?.Parent?.IsKind(SyntaxKind.PrimaryConstructorBaseType) is true;
```

Comment:
this is a bit of an odd expression given the name of hte local function.  IsIn implies that we're *anywhere* within the arg list.  but here it seems like it's asking about very specific topmost nodes in the arg list.

Reply:
my point is that the impl of this method doesn't match the name :)